standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2104,9 +2106,14 @@ public void truncateTable(String catName, String dbName, String tableName, List<
   private void truncateTableInternal(String catName, String dbName, String tableName,
       List<String> partNames, String validWriteIds, long writeId, boolean deleteData)
           throws TException {
+    EnvironmentContext context = new EnvironmentContext();
+    if (tableName.contains(".")) {
```

Comment:
should we use `HiveTableName.of(tableName)` instead?

Reply:
``HiveTableName`` is part of hive-exec, that ain't a dependency here :-(

Reply:
:(